types/scenegraph.d.ts

```diff
@@ -10,6 +10,122 @@ declare interface ScaleFactor {
     scaleY: number;
 }
 
+/**
+ * (**Since:** XD 29)
+ *
+ * Stores metadata accessible to multiple plugins, separated into silos by plugin ID. Your plugin can read & write the storage for its own plugin ID, but storage for other plugin IDs is *read-only*.
+ *
+ * Each per-plugin storage silo is a collection of key-value pairs. Keys and values must both be strings.
+ *
+ * *Each* scenenode has its own metadata storage, accessed via `SceneNode.sharedPluginData`. To store general metadata that is not specific to one scenenode, use `sharedPluginData` on the document's scenegraph root.
+ * @example ```js
+ * // This example shows how to save & retrieve rich JSON data in shared metadata storage.
+ // See below for simpler examples of using individual APIs.
+ const PLUGIN_ID = "<your manifest's plugin ID here>";
+ let richObject = {
+    list: [2, 4, 6],
+    name: "Hello world"
+ };
+ node.sharedPluginData.setItem(PLUGIN_ID, "richData", JSON.stringify(richObject));
+
+ // Later on...
+ // (This could be in a different plugin, if it passes the original plugin's ID here)
+ let jsonString = node.sharedPluginData.getItem(PLUGIN_ID, "richData");
+ if (jsonString) {  // may be undefined
+    let richObjectCopy = JSON.parse(jsonString);
+    console.log(richObjectCopy.list.length);  // 3
+ }
+ */
+declare interface PerPluginStorage {
+    /**
+     * Returns a map where key is plugin ID and value is a nested map containing all the shared metadata for that plugin ID (i.e. the result of calling `getForPluginId()` with that ID).
+     */
```

Comment:
This is missing part of the [current?] documentation as well as the example.

```suggestion
    /**
     * Returns a map where key is plugin ID and value is a nested map containing all the shared metadata for that plugin ID (i.e. the result of calling `getForPluginId()` with that ID).
     * 
     * This map is a clone of the stored metadata, so modifying it has no effect.
     * 
     * @example
     * ```
     * let allSharedMetadata = node.sharedPluginData.getAll();
     * console.log("Plugin A's 'foo' value:",
     *             allSharedMetadata["A"] && allSharedMetadata["A"].foo);
     * console.log("All of plugin B's shared metadata on this node:",
     *             allSharedMetadata["B"]);
     * console.log("List of plugins storing shared metadata on this node:",
     *             Object.keys(allSharedMetadata));
     * ```
     */
```